src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let symbal = '';
+    let currentSep = separator;
+
+    if (separator === null) {
+      currentSep = 'null';
+    }
```

Comment:
in my opinion that is unnecessary, you just can use an argument

Reply:
thanks